src/gui.ts

```diff
@@ -0,0 +1,280 @@
+import * as blessed from "blessed";
+import {
+  Cell,
+  cellHasStone,
+  GameState,
+  gameIsFinished,
+  updateGameState,
+  getWinner,
+} from "./gameState";
+import { UTF16_CODE_OF_LETTER_A } from "./utils";
+
+const CELL_WIDTH = 2;
+const CELL_HEIGHT = 1;
+const PADDING_WIDTH = 3;
+const PADDING_HEIGHT = 2;
+const RENDERED_STONE = "⬢";
+const RENDERED_NO_STONE = "⬡";
+const RENDERED_SPACE = " ";
+const TEXT_LEFT_OFFSET = -30;
+const HOVER_FG_COLOR = "#999999";
+
+export function initScreen(): blessed.Widgets.Screen {
+  const screen = blessed.screen({
+    smartCSR: true,
+  });
+  // Quit on Escape, q, or Control-C.
+  screen.key(["escape", "q", "C-c"], function (ch, key) {
+    return process.exit(0);
+  });
+  return screen;
+}
+
+export function renderBoardAndLoop(
+  gameState: GameState,
+  screen: blessed.Widgets.Screen
+) {
+  const boardLayout = createBoardOuterLayout(gameState);
+
+  // For each column
+  gameState.board[0].forEach((_, x) => {
+    // Create col header label
+    createColumnHeaderLabel(boardLayout, x);
+    // Create top edge
+    createBoardTopEdge(boardLayout, x);
+    // Create bottom edge
+    createBoardBottomEdge(boardLayout, gameState, x);
+  });
+
+  // For each row
+  gameState.board.forEach((row, y) => {
+    // Create row header label
+    createRowHeaderLabel(boardLayout, y);
+    // Create text element for each cell in a row
+    row.forEach((_, x) => {
+      boardLayout.append(createCell(gameState, screen, x, y));
+    });
+    // Create left edge
+    createBoardLeftEdge(boardLayout, y);
+    // Create right edge
+    createBoardRightEdge(boardLayout, y, gameState);
+  });
+
+  // Add game title label
+  boardLayout.append(createGameTitleLabel());
+
+  // Add player turn label
+  boardLayout.append(createCurrentPlayerMsg(gameState));
+
+  // Add msg to tell if so has won
+  if (gameIsFinished(gameState)) {
+    boardLayout.append(createWinMsg(gameState));
+    boardLayout.append(createHelpMsg(gameState));
+  }
+
+  screen.append(boardLayout);
+  screen.render();
+}
+
+function createGameTitleLabel() {
+  return blessed.text({
+    top: 0,
+    left: TEXT_LEFT_OFFSET,
+    content: "The Game of Hex",
```

Comment:
Nothing to change here but just a thought, we have more and more texts and I think that one day it will be useful to extract all these texts in separate files to allow to maintain them more easily